contentcuration/contentcuration/frontend/shared/data/resources.js

```diff
@@ -1088,90 +1091,94 @@ export const ContentNode = new Resource({
-        position = RELATIVE_TREE_POSITIONS.LAST_CHILD;
-      }
 
-      if (lft === null) {
-        return Promise.reject('new lft value evaluated to null');
-      }
+    return this.resolveTreeInsert(id, target, position, true, data => {
+      data.change.exclude_descendants = excluded_descendants;
 
-      let data = { parent, lft, changed: true };
-      let oldObj = null;
-      return this.table
-        .get(id)
-        .then(node => {
-          oldObj = node;
-          return this.table.update(id, data);
-        })
-        .then(() => {
-          // Set old parent to changed
-          if (oldObj.parent !== parent) {
-            return this.table.update(oldObj.parent, { changed: true });
-          }
-        })
-        .then(updated => {
-          if (updated) {
-            // Update succeeded
-            return { id, ...data };
-          }
-          // Update didn't succeed, this node probably doesn't exist, do a put instead,
-          // but need to add in other parent info.
-          return this.table.get(parent).then(parentNode => {
-            data = {
-              id,
-              parent,
-              lft,
-              root_id: parentNode.root_id,
-              changed: true,
-            };
-            return this.table.put(data).then(() => data);
-          });
-        })
-        .then(data => {
-          return db[CHANGES_TABLE].put({
-            key: id,
-            target,
-            position,
-            oldObj,
-            source: CLIENTID,
-            table: this.tableName,
-            type: CHANGE_TYPES.MOVED,
-          }).then(() => data);
-        });
+      // Ignore changes from this operation except for the
+      // explicit copy change we generate.
+      return this.transaction({ mode: 'rw', source: IGNORED_SOURCE }, CHANGES_TABLE, () => {
+        return this.tableCopy(data);
+      });
     });
   },
 
+  tableCopy({ node, parent, payload, change }) {
+    payload = {
+      ...node,
+      ...payload,
+      published: false,
+      // Placeholder node_id, we should receive the new value from backend result
+      node_id: uuid4(),
+      original_source_node_id: node.original_source_node_id || node.node_id,
+      source_channel_id: node.channel_id,
+      source_node_id: node.node_id,
+      channel_id: parent.channel_id,
+      root_id: parent.root_id,
+      // Set this node as copying until we get confirmation from the
+      // backend that it has finished copying
+      [COPYING_FLAG]: true,
+      // Set to null, but this should update to a task id to track the copy
+      // task once it has been initiated
+      [TASK_ID]: null,
+    };
+
+    // Manually put our changes into the tree changes for syncing table
+    return this.table
+      .put(payload)
+      .then(() => db[CHANGES_TABLE].put(change))
+      .then(() => payload);
+  },
+
   getAncestors(id) {
-    return this.table.get(id).then(node => {
+    return this.get(id).then(node => {
```

Comment:
We might want to revert this one, as the intention is to just get ancestors from the local table until we have to fetch from the backend with the request collection.